src/Core/tests/UnitTests/Views/ShapeTests.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.Maui.Controls;
+using Microsoft.Maui.Controls.Shapes;
+using Microsoft.Maui.Graphics;
+using Xunit;
+
+namespace Microsoft.Maui.UnitTests.Views
+{
+	[Category(TestCategory.Core, TestCategory.View)]
+	public class ShapeTests
+	{
+		[Fact]
+		public void ShapeFillSubscribed()
+		{
+			var fill = new SolidColorBrush { Color = Colors.Red };
+			var shape = new Rectangle { Fill = fill };
+
+			bool fired = false;
+			shape.PropertyChanged += (sender, e) =>
+			{
+				if (e.PropertyName == nameof(Shape.Fill))
+					fired = true;
+			};
+
+			fill.Color = Colors.Green;
+			Assert.True(fired, "PropertyChanged did not fire!");
+		}
+
+		[Fact]
+		public async Task ShapeFillDoesNotLeak()
+		{
+			var fill = new SolidColorBrush(Colors.Red);
+			var reference = new WeakReference(new Rectangle { Fill = fill });
+
+			await Task.Yield();
+			GC.Collect();
+			GC.WaitForPendingFinalizers();
+
+			Assert.False(reference.IsAlive, "Shape should not be alive!");
+		}
+
+		[Fact]
+		public void ShapeStrokeSubscribed()
+		{
+			var stroke = new SolidColorBrush { Color = Colors.Red };
+			var shape = new Rectangle { Stroke = stroke };
+
+			bool fired = false;
+			shape.PropertyChanged += (sender, e) =>
+			{
+				if (e.PropertyName == nameof(Shape.Stroke))
+					fired = true;
+			};
+
+			stroke.Color = Colors.Green;
+			Assert.True(fired, "PropertyChanged did not fire!");
```

Comment:
I think I found an issue in this pattern, if you change this to:
```suggestion
			await Task.Yield();
			GC.Collect();
			GC.WaitForPendingFinalizers();

			stroke.Color = Colors.Green;
			Assert.True(fired, "PropertyChanged did not fire!");
```
Does it fail? I found saving some delegates in a member variable fixes it, but looking how we can do it in a nicer way.

Reply:
Follow what I did here to get the above test to pass: https://github.com/dotnet/maui/pull/13997

Reply:
Right, the test was not passing in that case. 
Applied changes to save the EventHandlers in a member field.

![image](https://user-images.githubusercontent.com/6755973/225874648-2573d7fb-7d51-4410-b4bc-4182a5378fe0.png)
Now are passing (at least locally).